### PR TITLE
feat: Add search through resources functionality 

### PR DESCRIPTION
**Pull request by yahia3200**
## Demo

Peek 2023-07-22 21-20

## Changes Made
- Seprate Navbar into a seprate file 
- Implement search with debounce function 
- separate filtering and search logic in a custom react hook
- focus on search input when `ctrl` + `f` are pressed

## Additional Notes
For now, the search is done using the resource name only but this can easily be extended to different resource attributes

### CODE REVIEW

**Comment by mshudrak** [comment]
@yahia3200 could you please add a simple manual on how to run the tool in the description here? https://github.com/google/gcp_scanner#working-with-results. I want to run it to test and provide feedback.

**Comment by yahia3200** [comment]
@mshudrak for now to work with the vis tool you need node.js to be installed on your system and following this steps

`cd visualization_tool`
`npm i`
`npm run dev`

As this not the way that the user should use the tool, I didn't add these steps to the repo
In the future I will add a build step and bundle the tool with gcp scanner and use flask to start the tool. This is partially done but not complete yet

File: visualization_tool/src/components/Navbar/Navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      <header>
+        <img src={logo} className="logo" alt="GCP Scanner logo" />
+        <p>Scanner</p>
+      </header>
+      <div className="pages-nav">
+        <div className="links">
+          <a className="active" href="">
+            Resources
+          </a>
+          <a href="">IAM Policy</a>
```
Comment (mshudrak) [inline visualization_tool/src/components/Navbar/Navbar.tsx:44]: I know it was already reviewed and approved but why do we have IAM Policy as a separate tab?

**Comment by mshudrak** [comment]
Thanks. it works for me. I am trying to load test report but it shows just "no resource found". Am I doing something wrong? I left 2 comments/questions

File: visualization_tool/src/components/Navbar/Navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, []);
+
+  return (
+    <>
+      <header>
+        <img src={logo} className="logo" alt="GCP Scanner logo" />
+        <p>Scanner</p>
```
[the remark this answers is not in the document]
Comment (yahia3200) [reply visualization_tool/src/components/Navbar/Navbar.tsx]: Done

File: visualization_tool/src/components/Navbar/Navbar.tsx (hunk 1 above)
Comment (yahia3200) [reply visualization_tool/src/components/Navbar/Navbar.tsx:44]: This is just a place holder for other tabs in the feature and to show the design
I think we can remove it now

**Comment by yahia3200** [comment]
I saw the two comments but didn't see this one 
After you upload the file, did you click on the plus icon ?

**Comment by mshudrak** [comment]
All good, sometimes it is hard to see them :) Yes, I upload the file and clicked on plus button.

**Comment by mshudrak** [comment]
You can try it on your own. 
[test.zip](https://github.com/google/gcp_scanner/files/12174304/test.zip)

**Comment by yahia3200** [comment]
I made a mistake in renaming the resources in the tool and the final I'm using for demo. The tool now only parses the `compute_instances` and `compute_disks` resources not `storage_buckets`.

I made a commit to fix this issue and use names `Compute Instance` and `Compute Disk` instead of `GCE` and `GCS`
In my plan that In the next PR the tool should parse all resources in the `light_version_scan_schema`

**Review by mshudrak — APPROVED** [review]

**Pull request merged**